cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/validator/CoreValidator.java

```diff
@@ -0,0 +1,311 @@
+    }
+
+    /**
+     * Checks if the value can safely be converted to a int primitive.
+     */
+    public static boolean isInt(String input) {
+      return GenericValidator.isInt(input);
+    }
+
+    /**
+     * Checks if the value can safely be converted to a long primitive.
+     */
+    public static boolean isLong(String input) {
+      return GenericValidator.isLong(input);
+    }
+
+    /**
+     * Checks if the value can safely be converted to a short primitive.
+     */
+    public static boolean isShort(String input) {
+      return GenericValidator.isShort(input);
+    }
+
+    /**
+     * Checks if a field is a valid URL address.
+     */
+    public static boolean isUrl(String url) {
+      return URL_VALIDATOR.isValid(url);
+    }
+
+    /**
+     * Checks if the value matches the regular expression.
+     */
+    public static boolean matchRegex(String pattern, String input) {
+      return new RegexValidator(pattern).isValid(input);
+    }
+
+    /**
+     * Checks if the value's adjusted length is less than or equal to the max.
+     */
+    public static boolean maxLength(String input, int maxLength) {
+      return GenericValidator.maxLength(input, maxLength);
+    }
+
+    /**
+     * Checks if the value is less than or equal to the max.
+     */
+    public static boolean maxValue(double val, double maxVal) {
+      return GenericValidator.maxValue(val, maxVal);
+    }
+
+    /**
+     * Checks if the value is less than or equal to the max.
+     */
+    public static boolean maxValue(long val, long maxVal) {
+      return GenericValidator.maxValue(val, maxVal);
+    }
+
+    /**
+     * Checks if the value is less than or equal to the max.
+     */
+    public static boolean maxValue(int val, int maxVal) {
+      return GenericValidator.maxValue(val, maxVal);
+    }
+
+    /**
+     * Checks if the value is less than or equal to the max.
+     */
+    public static boolean maxValue(float val, float maxVal) {
+      return GenericValidator.maxValue(val, maxVal);
+    }
+
+    /**
+     * Checks if the value is greater than or equal to the min.
+     */
+    public static boolean minValue(double val, double minVal) {
+      return GenericValidator.minValue(val, minVal);
+    }
+
+    /**
+     * Checks if the value is greater than or equal to the min.
+     */
+    public static boolean minValue(long val, long minVal) {
+      return GenericValidator.minValue(val, minVal);
+    }
+
+    /**
+     * Checks if the value is greater than or equal to the min.
+     */
+    public static boolean minValue(int val, int minVal) {
+      return GenericValidator.minValue(val, minVal);
+    }
+
+    /**
+     * Checks if the value is greater than or equal to the min.
+     */
+    public static boolean minValue(float val, float minVal) {
+      return GenericValidator.minValue(val, minVal);
+    }
+
+    /**
+     * Checks if the value's adjusted length is greater than or equal to the min.
```

Comment:
Checks if the input value's adjusted length is greater than or equal to the minimum specified by length.